### PR TITLE
find the branch point properly

### PR DESCRIPTION
**Pull request by bhcleek**
Calculate the branch point by examining the branch's history instead of relying on finding the merge base from base branch so that merging the base branch into the branch does not cause gta to disregard changes made on the branch prior to said merge.

### CODE REVIEW

**Review by nesv — APPROVED** [review]

**Review by rmg — DISMISSED** [review]
LGTM :shipit: 

Do the existing tests cover this well enough?

**Comment by bhcleek** [comment]
@rmg they did not cover them well enough. I've added an integration test that covers this case. I rebased so that the test is added in the first commit with the changes that you and Nick already approved as the second commit. The second commit is unaltered from what you've already approved. Ordering it this way makes it easy to checkout the commit that adds the test to verify that it fails without the substantive changes in the second commit.

File: gtaintegration/gtaintegration_test.go [hunk 1]
```diff
@@ -427,6 +427,132 @@ func TestPackageRemoval_MovePackage(t *testing.T) {
...
+
+		t.Logf("\n%s", out)
+	}
+
+	if _, err := runGit(ctx, ".", "commit", "-a", "-m", "delete some stuff"); err != nil {
+		t.Fatal(err)
+	}
+
+	// merge the base branch. After this, the branch topology should look like this:
+	// * (topic) merge
```
Comment (rmg) [inline gtaintegration/gtaintegration_test.go:472]: Is this enough history to verify that this works better than `merge-base`? Like a commit to `master` and `topic` _after_ the merge?
```
* b91ae25 (HEAD -> main) add main
| * 075acfc (topic) add topic
| *   01c87c7 Merge branch 'main' into topic
| |\
| |/
|/|
* | 5220151 delete some stuff
| * 9dc58c8 moved some stuff
|/
* 67d81d6 initial commit
```

Comment (bhcleek) [reply gtaintegration/gtaintegration_test.go:472]: It is enough, yes. The git differ used to use `base...branch`, and the tests demonstrate that was insufficient prior to the changes in the second commit.

**Review by rmg — APPROVED** [review]
LGTM :shipit:

**Pull request merged**